sx.el

```diff
@@ -259,6 +259,23 @@ whenever BODY evaluates to nil."
        :filter (lambda (&optional _)
                  (when (progn ,@body) ,def)))))
 
+(defmacro sx--create-comparator (name doc compare-func get-func)
+  "Define a new comparator called NAME with documentation DOC.
+COMPARE-FUNC is a function that takes the return value of
+GET-FUNC and performs the actual comparison."
+  (declare (indent 1) (doc-string 2))
+  (let ((gf (intern (format " %S--get-prop-function" name)))
+        (cf (intern (format " %S--compare-function"  name))))
+    ;; Leading space to hide from completion systems
+    `(progn
+       ;; In using `defalias', the macro supports both function
+       ;; symbols and lambda expressions.
+       (defalias ',gf ,get-func)
+       (defalias ',cf ,compare-func)
+       (defun ,name (a b)
+         ,doc
+         (,cf (,gf a) (,gf b))))))
```

Comment:
If we really don't want to use `<` instead of `#'<`, we can use `(funcall ,compare-func ...` instead of defining defaliases.